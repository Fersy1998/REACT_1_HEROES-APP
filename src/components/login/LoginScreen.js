import React, { useContext } from 'react'
import { useNavigate } from 'react-router-dom'
import { AuthContext } from '../../auth/authContext'
import { types } from '../../types/types'

export const LoginScreen = () => {
    
    const navigate=useNavigate()
    
    //de context sacamos el dispatch
    //const context = useContext(AuthContext)
    const {dispatch} = useContext(AuthContext)
    
    const handleLogin=()=>{
        const last_path=localStorage.getItem('last_path') || ('/search');
        const action={
          type: types.login,
          payload:{
            name:'Fersy'
          }
        
        }
        dispatch(action);
      
        navigate(last_path, {
          replace:true
        }) ;
    }
  return (
    <div className='container mt-5'>
        <h5>Login Screen</h5>   <hr />
        <button className='btn btn-primary' onClick={handleLogin}>Login</button>
    </div>
  )
}
