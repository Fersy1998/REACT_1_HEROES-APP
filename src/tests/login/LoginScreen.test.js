import { LoginScreen } from "../../components/login/LoginScreen";
import { mount } from 'enzyme';
import { AuthContext } from "../../auth/authContext";
import { MemoryRouter, useNavigate } from "react-router-dom";
import { types } from "../../types/types";

const mockUseNavigate=jest.fn();
jest.mock('react-router-dom', ()=>({
    ...jest.requireActual('react-router-dom'),
    useNavigate:()=>mockUseNavigate
}))

describe('Pruebas en LoginScreen', ()=>{

    test('should match the snapshot', () => { 
    /*
        const contextValue={
        }*/
        const wrapper=mount(
        <AuthContext.Provider value={ {user:{logged:false}}} >
            <MemoryRouter initialEntries={['/login']}>
                <LoginScreen />
            </MemoryRouter>
        </AuthContext.Provider>)
        expect(wrapper).toMatchSnapshot();
    
    })
    const dispatch=jest.fn();
    test('should match the snapshot', () => { 
            
            const action={
                type:types.login,
                payload:{
                    name:'Fersy'
                }
            }
            const wrapper=mount(
            <AuthContext.Provider value={ {user:{logged:false}, dispatch}} >
                <MemoryRouter initialEntries={['/login']}>
                    <LoginScreen />
                </MemoryRouter>
            </AuthContext.Provider>)
            wrapper.find('button').simulate('click');
            expect(dispatch).toHaveBeenCalledWith(action);
            expect(mockUseNavigate).toHaveBeenCalledWith('/search', {replace:true});
            
            
        })
    


})