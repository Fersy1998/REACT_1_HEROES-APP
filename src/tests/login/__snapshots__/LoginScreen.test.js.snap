// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginScreen should match the snapshot 1`] = `
<div
  className="container mt-5"
>
  <h5>
    Login Screen
  </h5>
     
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
